UnityProject/Assets/Scripts/Clothing/Eyewear/NightVisionGoggles.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using CameraEffects;
+
+public class NightVisionGoggles : NetworkBehaviour, IServerInventoryMove
+{
+	[SerializeField, Tooltip("How far the player will be able to see in the dark while he has the goggles on.")]
+	private Vector3 nightVisionVisibility;
+
+	[SerializeField, Tooltip("The default minimal visibility size.")]
+	private Vector3 defaultVisionVisibility;
+
+	
+	private void OnStartClient()
+	{
+		if(PlayerManager.LocalPlayerScript != null)
+		{
+			var item = PlayerManager.LocalPlayerScript.Equipment.GetClothingItem(NamedSlot.eyes).GameObjectReference;
+			if(item == gameObject) {enableEffect(true);}
+		}
+	}
+
+	public void OnInventoryMoveServer(InventoryMove info)
+		{
+			RegisterPlayer registerPlayer = info.ToRootPlayer;
+
+			if (info.ToSlot != null && info.ToSlot?.NamedSlot != null)
+			{
+
+				if (registerPlayer != null && info.ToSlot.NamedSlot == NamedSlot.eyes)
+				{
+					TargetOnWearing(registerPlayer.connectionToClient);
+				}
+			}
+
+			if (info.FromSlot != null && info.FromSlot?.NamedSlot != null)
+			{
+				if (registerPlayer != null && info.FromSlot.NamedSlot == NamedSlot.eyes)
+				{
+					TargetOnTakingOff(registerPlayer.connectionToClient);
+				}
+			}
+		}
+	
+	[TargetRpc]
+	private void TargetOnTakingOff(NetworkConnection target)
+		{
+			enableEffect(false);
+		}
+
+	[TargetRpc]
+	private void TargetOnWearing(NetworkConnection target)
+		{
+			enableEffect(true);
+		}
+	
+	private void enableEffect(bool check)
+	{
+		var camera = Camera.main;
+		if(check == true)
+		{
+			camera.GetComponent<CameraEffectControlScript>().ToggleNightVisionEffectState();
+			camera.GetComponent<CameraEffectControlScript>().AdjustPlayerVisibility(nightVisionVisibility);
+		}
+		else
+		{
+			camera.GetComponent<CameraEffectControlScript>().ToggleNightVisionEffectState();
+			camera.GetComponent<CameraEffectControlScript>().AdjustPlayerVisibility(defaultVisionVisibility);
+		}
```

Comment:
Can you remove these indents

Reply:
I can't find them, vscode is also starting to annoy me about this problem.

Reply:
i did it for you, strange the vscode wasnt showing them